db/commitlog/commitlog.cc

```diff
@@ -1375,11 +1375,21 @@ db::commitlog::segment_manager::segment_manager(config c)
     , max_mutation_size(max_size >> 1)
     , max_disk_size(size_t(std::ceil(cfg.commitlog_total_space_in_mb / double(smp::count))) * 1024 * 1024)
     // our threshold for trying to force a flush. needs heristics, for now max - segment_size/2.
-    , disk_usage_threshold(cfg.commitlog_flush_threshold_in_mb.has_value() 
-        ? size_t(std::ceil(*cfg.commitlog_flush_threshold_in_mb / double(smp::count))) * 1024 * 1024 
-        : (max_disk_size -
-            (max_disk_size >= (max_size*2) ? max_size
-                : (max_disk_size > (max_size/2) ? (max_size/2) : max_disk_size/3))))
+    , disk_usage_threshold([] (config &cfg, const uint64_t &max_disk_size, const uint64_t &max_size) {
```

Comment:
Why not just `[&]` capture?

Reply:
No good reason :slightly_smiling_face: 
Fixed